src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1496,6 +1496,17 @@
 
   {::vacuum-analyze #{"factsets"}})
 
+(defn resource-events-pk []
+  (jdbc/do-commands
+   "alter table resource_events add column event_hash bytea"
+
+   "update resource_events re
+    set event_hash = digest(concat(re.report_id, re.resource_type, re.resource_title, coalesce(re.property, ''))::text, 'sha1'::text)"
+
+   "alter table resource_events add primary key (event_hash)")
+
+  {::vaccum-analyze #{"resource_events"}})
```

Comment:
I have noticed a few of our migrations end with this vacuum analyze, what does it do?

Reply:
It makes the database update the table statistics used in the query planner. It's important for ensuring stable performance of joins and index usage and such. https://www.postgresql.org/docs/9.6/sql-vacuum.html